asmcomp/arm64/emit.mlp

```diff
@@ -38,7 +38,7 @@ let reg_trap_ptr = phys_reg 23
 let reg_alloc_ptr = phys_reg 24
 let reg_alloc_limit = phys_reg 25
 let reg_tmp1 = phys_reg 26
-let reg_x15 = phys_reg 15
+let reg_x18 = phys_reg 28
```

Comment:
Please don't use register 18 ever.  Using r8 will work fine.

Reply:
Ok, I thought it was okay as it is safe on Linux, but I agree that this is a bad idea, will address it.